### ISSUE TITLE
TypeError: persist() missing 1 required positional argument: 'model_dir'

### ISSUE THREAD

**Issue by kaiwk** [RasaHQ/rasa_nlu#1777]
**Rasa NLU version**: 0.14.4

**Operating system**: OS X

**Content of model configuration file**:
```yml
language: "zh"

pipeline:
    - name: "nlp_mitie"
      model: "data/total_word_feature_extractor_zh.dat"
    - name: "tokenizer_jieba_search.JiebaTokenizer"
      dictionary_path: "data/jieba_dict.txt"
    - name: "ner_mitie"
    - name: "ner_synonyms"
    - name: "intent_entity_featurizer_regex"
    - name: "intent_featurizer_mitie"
    - name: "intent_classifier_sklearn"
```

**Issue**:
I substituted `tokenizer_jieba` with the customized one which is I copied from here (https://raw.githubusercontent.com/RasaHQ/rasa_nlu/fff482663cd33ec16b16a39847e58884f4b4c9a2/rasa_nlu/tokenizers/jieba_tokenizer.py) without chaning anything, and renamed it to `tokenizer_jieba_search.py`.

It gives error log below:

```
Traceback (most recent call last):
  File "/Users/kai/.pyenv/versions/3.6.6/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/Users/kai/.pyenv/versions/3.6.6/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/kai/.virtualenvs/rasa-play-re5z2bZX/lib/python3.6/site-packages/rasa_nlu/train.py", line 184, in <module>
    num_threads=cmdline_args.num_threads)
  File "/Users/kai/.virtualenvs/rasa-play-re5z2bZX/lib/python3.6/site-packages/rasa_nlu/train.py", line 160, in do_train
    fixed_model_name)
  File "/Users/kai/.virtualenvs/rasa-play-re5z2bZX/lib/python3.6/site-packages/rasa_nlu/model.py", line 233, in persist
    update = component.persist(dir_name)
TypeError: persist() missing 1 required positional argument: 'model_dir'
```
If I use `tokenizer_jieba`, it would works fine.

Since I didn't change the tokenizer code (except its file name), I thought it will works. Was I missed something?

**Comment by akelad** [comment]
Thanks for raising this issue, @MetcalfeTom  will get back to you about it soon.

**Comment by akelad** [comment]
@erohmensing can you take a look instead?

**Comment by erohmensing** [comment]
@kaiwk It looks like the code you copied is out of date with changes that went into 0.14.4 -- you'll notice [here](https://github.com/RasaHQ/rasa_nlu/blob/0.14.4/rasa_nlu/tokenizers/jieba_tokenizer.py#L115) that the `persist()` method you replaced takes 2 arguments, while the code you copied takes 3. 

Where did that raw code hunk come from? It looks like it's essentially the master version -- you can't really pull new code into old versions and expect it to work. What are you trying to achieve with it that you can't currently?

**Comment by kaiwk** [comment]
@erohmensing You are right, I used the wrong version. After upgrading to:

>rasa-core           0.13.5                                                              
rasa-core-sdk       0.12.2                                                              
rasa-nlu            0.14.6

everything works.

[Jieba](https://github.com/fxsjy/jieba) itself support different tokenize mode. I tried add extra configuration parameters to jieba tokenizer.  If possible, I would like make a PR for this.

Thank you for the help!
